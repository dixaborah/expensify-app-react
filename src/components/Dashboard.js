import React from 'react'
import ExpenseList from '../views/ExpenseList'
import ExpenseListFilters from '../views/ExpenseListFilters'

const Dashboard = () => (
    <div>
        <ExpenseListFilters />
        <ExpenseList />
    </div>
)

export default Dashboard