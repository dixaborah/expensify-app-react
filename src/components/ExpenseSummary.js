import React from 'react'
import numeral from 'numeral'
import { connect } from 'react-redux'

import SelectExpenseTotal from '../selectors/ExpenseTotal'
import SelectExpenses from '../selectors/ExpenseTotal'

export const ExpenseSummary = ({ expenseCount, expenseTotal }) => {
    const expenseWord = expenseCount === 1 ? 'expense' : 'expenses'
    const formattedExpenseTotal = numeral(expenseTotal / 100).format('$0,0.00')
    return (
        <h1>Viewing {expenseCount} {expenseWord} totalling {formattedExpenseTotal}</h1>
    )
}

const mapStateToProps = (state) => {
    const visibleExpenses = SelectExpenses(state.expenses, state.filters)

    return {
        expenseCount: visibleExpenses.length,
        expenseTotal: SelectExpenseTotal(visibleExpenses)
    }
}

export default connect(mapStateToProps)(ExpenseSummary)