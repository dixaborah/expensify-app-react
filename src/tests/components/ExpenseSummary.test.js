import React from 'react'
import { shallow } from 'enzyme'
import { ExpenseSummary } from '../../components/ExpenseSummary'

test('Should', () => {
    const wrapper = shallow(<ExpenseSummary expenseCount='5' expenseTotal='7755' />)
    expect(wrapper).toMatchSnapshot()
})