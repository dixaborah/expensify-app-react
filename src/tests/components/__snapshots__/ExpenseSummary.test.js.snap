// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should 1`] = `
<h1>
  Viewing 
  5
   
  expenses
   totalling 
  $77.55
</h1>
`;
